store/tikv/region_cache.go

```diff
@@ -329,6 +335,80 @@ func (c *RegionCache) GetRPCContext(bo *Backoffer, id RegionVerID, replicaRead k
 	}, nil
 }
 
+// GetFlashRPCContext returns RPCContext for a region must access flash store. If it returns nil, the region
+// must be out of date and already dropped from cache or no flash store found.
+func (c *RegionCache) GetFlashRPCContext(bo *Backoffer, id RegionVerID) (*RPCContext, error) {
+	ts := time.Now().Unix()
+
+	cachedRegion := c.getCachedRegionWithRLock(id)
+	if cachedRegion == nil {
+		return nil, nil
+	}
+	if !cachedRegion.checkRegionCacheTTL(ts) {
+		return nil, nil
+	}
+
+	regionStore := cachedRegion.getStore()
+
+	notFlashCnt := 0
+	for i, store := range regionStore.stores {
+		state := store.getResolveState()
+		switch state {
```

Comment:
the following code is duplicated with `getStoreAddr()`

Reply:
Not all the same. The `state` is `NeedCheck`, `store` should be `reResolved`.

Reply:
But we can also reuse the `getStoreAddr()`.